File: 0769-max-chunks-to-make-sorted/0769-max-chunks-to-make-sorted.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var maxChunksToSorted = function(arr) {
    let maxx = 0;
    let chunks = 0;
    for(let i=0; i<arr.length; i++){
        maxx = Math.max(arr[i],maxx);
        if(maxx == i){
            chunks++;
        } 
    }
    return chunks;
};

File: 0769-max-chunks-to-make-sorted/0769-max-chunks-to-make-sorted.kt
class Solution {
    fun maxChunksToSorted(arr: IntArray): Int {
        var maxSoFar = 0 
        var chunks = 0   
        for (i in arr.indices) {
            maxSoFar = maxOf(maxSoFar, arr[i]) 
            if (maxSoFar == i) {              
                chunks++
            }
        }
        return chunks
    }
}